examples/vae/train_vae.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+"""Chainer example: train a VAE on MNIST
+"""
+from __future__ import print_function
+import argparse
+
+import matplotlib.pyplot as plt
+import numpy as np
+import six
+
+import chainer
+from chainer import computational_graph
+from chainer import cuda
+from chainer import optimizers
+from chainer import serializers
+
+import data
+import net
+
+parser = argparse.ArgumentParser(description='Chainer example: MNIST')
+parser.add_argument('--initmodel', '-m', default='',
+                    help='Initialize the model from given file')
+parser.add_argument('--resume', '-r', default='',
+                    help='Resume the optimization from snapshot')
+parser.add_argument('--gpu', '-g', default=-1, type=int,
+                    help='GPU ID (negative value indicates CPU)')
+parser.add_argument('--epoch', '-e', default=100, type=int,
+                    help='number of epochs to learn')
+parser.add_argument('--dimz', '-z', default=20, type=int,
+                    help='dimention of encoded vector')
+parser.add_argument('--batchsize', '-b', type=int, default=100,
+                    help='learning minibatch size')
+args = parser.parse_args()
+
+batchsize = args.batchsize
+n_epoch = args.epoch
+n_latent = args.dimz
+
+print('GPU: {}'.format(args.gpu))
+print('# dim z: {}'.format(args.dimz))
+print('# Minibatch-size: {}'.format(args.batchsize))
+print('# epoch: {}'.format(args.epoch))
+print('')
+
+# Prepare dataset
+print('load MNIST dataset')
+mnist = data.load_mnist_data()
+mnist['data'] = mnist['data'].astype(np.float32)
+mnist['data'] /= 255
+mnist['target'] = mnist['target'].astype(np.int32)
+
+N = 60000
+x_train, x_test = np.split(mnist['data'],   [N])
+y_train, y_test = np.split(mnist['target'], [N])
+N_test = y_test.size
+
+# Prepare VAE model, defined in net.py
+model = net.VAE(784, n_latent, 500)
+if args.gpu >= 0:
+    cuda.get_device(args.gpu).use()
+    model.to_gpu()
+xp = np if args.gpu < 0 else cuda.cupy
+
+# Setup optimizer
+optimizer = optimizers.Adam()
+optimizer.setup(model)
+
+# Init/Resume
+if args.initmodel:
+    print('Load model from', args.initmodel)
+    serializers.load_npz(args.initmodel, model)
+if args.resume:
+    print('Load optimizer state from', args.resume)
+    serializers.load_npz(args.resume, optimizer)
+
+# Learning loop
+for epoch in six.moves.range(1, n_epoch + 1):
+    print('epoch', epoch)
+
+    # training
+    perm = np.random.permutation(N)
+    sum_loss = 0       # total loss
+    sum_rec_loss = 0   # reconstruction loss
+    for i in six.moves.range(0, N, batchsize):
+        x = chainer.Variable(xp.asarray(x_train[perm[i:i + batchsize]]))
+        optimizer.update(model.get_loss_func(), x)
+        if epoch == 1 and i == 0:
+            with open('graph.dot', 'w') as o:
+                g = computational_graph.build_computational_graph(
+                    (model.loss, ), remove_split=True)
```

Comment:
`remove_split` option is left just for backward compatibility. So it is better to remove it.